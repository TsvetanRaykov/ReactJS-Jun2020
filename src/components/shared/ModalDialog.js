import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import Modal from '@material-ui/core/Modal'
import Backdrop from '@material-ui/core/Backdrop'
import Fade from '@material-ui/core/Fade'
import { Button, Box } from '@material-ui/core'

const useStyles = makeStyles((theme) => ({
	modal: {
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'center',
	},
	paper: {
		backgroundColor: theme.palette.background.paper,
		border: '2px solid #000',
		boxShadow: theme.shadows[5],
		padding: theme.spacing(2, 4, 3),
	},
}))

export default function ModalDialog(props) {
	const { open, handleYes, handleNo, title, message } = props

	const classes = useStyles()

	return (
		<div>
			<Modal
				aria-labelledby='modal-dialog-title'
				aria-describedby='modal-dialog-description'
				className={classes.modal}
				open={open}
				closeAfterTransition
				BackdropComponent={Backdrop}
				BackdropProps={{
					timeout: 500,
				}}
				disableEscapeKeyDown
				onClose={(event, reason) => {
					if (reason === 'backdropClick') {
						return false
					}
				}}
			>
				<Fade in={open}>
					<div className={classes.paper}>
						<h2 id='modal-dialog-title'>{title}</h2>
						<p
							id='modal-dialog-description'
							dangerouslySetInnerHTML={{ __html: message }}
						></p>

						<Box display='flex' mt={2} justifyContent='space-around'>
							<Button autoFocus onClick={() => handleNo()} color='primary'>
								Cancel
							</Button>
							<Button onClick={() => handleYes()} color='primary'>
								Ok
							</Button>
						</Box>
					</div>
				</Fade>
			</Modal>
		</div>
	)
}
