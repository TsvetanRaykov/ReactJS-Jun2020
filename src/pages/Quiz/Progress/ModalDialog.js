import React, { useState } from 'react'
import { makeStyles } from '@material-ui/core/styles'
import Modal from '@material-ui/core/Modal'
import Backdrop from '@material-ui/core/Backdrop'
import Fade from '@material-ui/core/Fade'

const useStyles = makeStyles((theme) => ({
	modal: {
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'center',
	},
	paper: {
		backgroundColor: theme.palette.background.paper,
		border: '2px solid #000',
		boxShadow: theme.shadows[5],
		padding: theme.spacing(2, 4, 3),
	},
}))

export default function ModalDialog() {
	const classes = useStyles()
	const [open, setOpen] = useState(false)

	const handleOpen = () => {
		setOpen(true)
	}

	const handleClose = () => {
		setOpen(false)
	}

	return (
		<div>
			<button type='button' onClick={handleOpen}>
				modal-dialog
			</button>
			<Modal
				aria-labelledby='modal-dialog-title'
				aria-describedby='modal-dialog-description'
				className={classes.modal}
				open={open}
				onClose={handleClose}
				closeAfterTransition
				BackdropComponent={Backdrop}
				BackdropProps={{
					timeout: 500,
				}}
			>
				<Fade in={open}>
					<div className={classes.paper}>
						<h2 id='modal-dialog-title'>Title</h2>
						<p id='modal-dialog-description'>text</p>
					</div>
				</Fade>
			</Modal>
		</div>
	)
}
