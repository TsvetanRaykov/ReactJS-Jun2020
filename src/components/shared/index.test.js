import React from 'react'
import renderer from 'react-test-renderer'
import TestWrapper from '../../utils/test-wrapper'
import ModalDialog from './ModalDialog'

jest.mock('react', () => ({
	...jest.requireActual('react'),
	useLayoutEffect: jest.requireActual('react').useEffect,
}))

describe('<ModalDialog />', () => {
	it('Should match the snapshot', () => {
		const wrapper = renderer.create(
			<TestWrapper>
				<ModalDialog open={false} />
			</TestWrapper>
		)
		expect(wrapper).toMatchSnapshot()
	})
})
