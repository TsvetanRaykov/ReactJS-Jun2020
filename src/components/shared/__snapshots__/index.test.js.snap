// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalDialog /> Should match the snapshot 1`] = `<div />`;
